internal/form/yes_no_form.go

```diff
@@ -11,22 +11,26 @@ type YesNoForm struct {
 	YesNo      YesNo
 	Error      error
 	ErrorLabel string
+	Options    YesNoOptions
+	FieldName  string
 }
 
 func ReadYesNoForm(r *http.Request, errorLabel string) *YesNoForm {
-	yesNo, err := ParseYesNo(PostFormString(r, "yes-no"))
+	form := NewYesNoForm(YesNoUnknown)
 
-	return &YesNoForm{
-		YesNo:      yesNo,
-		Error:      err,
-		ErrorLabel: errorLabel,
-	}
+	yesNo, err := ParseYesNo(PostFormString(r, form.FieldName))
```

Comment:
if you moved this line up you could do `form := NewYesNoForm(yesNo)`

Reply:
I could, but then we'd have to reference the field name directly rather than via the Form object. While that's not too terrible given where we are referencing it, it does break the encapsulation. What do you think?

Reply:
couldn't you use `FieldNames.YesNo`?